src/lcf/context.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * This file is part of liblcf. Copyright (c) 2020 liblcf authors.
+ * https://github.com/EasyRPG/liblcf - https://easyrpg.org
+ *
+ * liblcf is Free/Libre Open Source Software, released under the MIT License.
+ * For the full copyright and license information, please view the COPYING
+ * file that was distributed with this source code.
+ */
+
+#ifndef LCF_CONTEXT_H
+#define LCF_CONTEXT_H
+
+#include "string_view.h"
+
+namespace lcf {
+
+/**
+ * Context information for lcf fields
+ * @tparam T field type
+ */
+template <typename T>
+struct Context {
+	/** Context of the parent (nullptr when no parent) */
+	Context<T>* parent = nullptr;
+	/** Object instance (cast to appropriate RPG-type */
+	void* obj = nullptr;
```

Comment:
We're already passing obj to the function directly, why do we need this void pointer?

Reply:
This is used when you want to reference the instance of the parent object.
E.g. when I parse a map and the parser is on the string of an event command I can access the "lcf::rpg::Event" with 
``auto* event = reinterpret_cast<lcf::rpg::Event*>(ctx.parent->parent->obj)``

Reply:
Ok, I think I can add a little more template fairy dust and get rid of the need for the reinterpret_cast here. Do we want that?

Reply:
Well I couldn't figure out the template magic to get rid of the void* without making the API too ugly. But feel free to improve it :).